diff -ru SDL2-2.0.9/CMakeLists.txt SDL2-2.0.9_ori/CMakeLists.txt
--- SDL2-2.0.9_ori/CMakeLists.txt   2018-10-31 16:07:22.000000000 +0100
+++ SDL2-2.0.9/CMakeLists.txt       2019-06-29 21:02:06.084954500 +0200
@@ -3,7 +3,7 @@
 endif()

 cmake_minimum_required(VERSION 2.8.11)
-project(SDL2 C)
+project(SDL2)

 # !!! FIXME: this should probably do "MACOSX_RPATH ON" as a target property
 # !!! FIXME:  for the SDL2 shared library (so you get an
@@ -873,7 +873,7 @@
     set(HAVE_SDL_VIDEO TRUE)

     # Core stuff
-    find_library(ANDROID_DL_LIBRARY dl)
+    set(ANDROID_DL_LIBRARY libdl.so)
     find_library(ANDROID_LOG_LIBRARY log)
     find_library(ANDROID_LIBRARY_LIBRARY android)
     list(APPEND EXTRA_LIBS ${ANDROID_DL_LIBRARY} ${ANDROID_LOG_LIBRARY} ${ANDROID_LIBRARY_LIBRARY})
@@ -1734,7 +1734,11 @@
 add_library(SDL2main STATIC ${SDLMAIN_SOURCES})
 target_include_directories(SDL2main PUBLIC "$<BUILD_INTERFACE:${SDL2_SOURCE_DIR}/include>" $<INSTALL_INTERFACE:include/SDL2>)
 set(_INSTALL_LIBS "SDL2main")
-if (NOT ANDROID)
+if (ANDROID)
+  add_library(hidapi SHARED src/hidapi/android/hid.cpp)
+  target_link_libraries(hidapi log)
+  set(_INSTALL_LIBS "hidapi" ${_INSTALL_LIBS})
+else()
   set_target_properties(SDL2main PROPERTIES DEBUG_POSTFIX ${SDL_CMAKE_DEBUG_POSTFIX})
 endif()

